### PR TITLE
Featured Apps api utils

### PR DESCRIPTION
**Pull request by m-aboelenein**
# 🔘 PR Type

What kind of change does this PR introduce?

- [x] Enhancement

# 📜 Background

Provide a brief explanation of why this pull request is needed. Include the problem you are solving or the functionality you are adding. Reference any related issues.

Issue Link: #[[ENG-3261](https://linear.app/xverseapp/issue/ENG-3261/%5Bexpl-mob-1%5D-xverse-featured-apps-api)]
Context Link (if applicable):

# 🔄 Changes

Does this PR introduce a breaking change?

- [ ] Yes, Incompatible API changes
- [x] No, Adds functionality (backwards compatible)
- [ ] No, Bug fixes (backwards compatible)

Changes:

- Added the `getFeaturedDapps` method which fetches the list of featured dapps

# 🖼 Screenshot / 📹 Video

# ✅ Review checklist

Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

**Review by jordankzf — APPROVED** [review]

File: types/api/xverse/appConfig.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export interface FeaturedDapp {
+  name: string;
+  url: string;
+  icon: string;
+  description: string;
+  order: number;
+  banner?: string;
+}
+
+export interface AppConfig {
```
Comment (victorkirov) [inline types/api/xverse/appConfig.ts:19]: nit: prefer types over interfaces

File: types/api/xverse/appConfig.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export interface FeaturedDappOld {
+  name: string;
+  url: string;
+  image: string;
+  description: string;
+  order: number;
+  banner?: string;
+}
+
+export interface FeaturedDapp {
```
[the remark this answers is not in the document]
Comment (dhriaznov) [reply types/api/xverse/appConfig.ts:10]: There's more context on the API PR: https://github.com/secretkeylabs/xverse-api/pull/224#issuecomment-1857675358

File: api/xverse.ts [hunk 3]
```diff
@@ -304,10 +307,16 @@ export async function getInscription(
   return response.data;
 }
 
-export async function getAppConfig(network: NetworkType) {
+export async function getAppConfig(network: NetworkType): Promise<AppConfig> {
   const appConfigUrl = `${XVERSE_API_BASE_URL(network)}/v1/app-config`;
-  const appConfig = await axios.get(appConfigUrl);
-  return appConfig;
+  const response = await axios.get(appConfigUrl);
+  return response.data.appConfig;
```
Comment (victorkirov) [inline api/xverse.ts:313]: This looks like a breaking change 👀

File: api/xverse.ts [hunk 4]
```diff
@@ -304,10 +307,16 @@ export async function getInscription(
...
+}
+
+export async function getFeaturedDapps(sectionsEnabled = false): Promise<(FeaturedDapp | FeaturedDappOld)[]> {
```
Comment (victorkirov) [inline api/xverse.ts:316]: If `FeaturedDapp | FeaturedDappOld` is due to an api schema change, then we should be changing the end point to a v2 endpoint

File: types/api/xverse/appConfig.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (victorkirov) [reply types/api/xverse/appConfig.ts:10]: That should've gone to a v2 endpoint :/ We don't want an API end point to return different schemas. Could we make the change in the API repo and change this to only have the new schema and point to the v2 endpoint?

[the remark this answers is not in the document]
Comment (victorkirov) [reply types/api/xverse/appConfig.ts:10]: It also looks like that PR went into the dev branch 👀 That is currently only hosted on render and the change wouldn't have made it into api-3.xveres.app. It needs to go into dev-k8s for that. That's our main branch on the API until we migrate api.xverse.app to k8s.

**Review by victorkirov — CHANGES_REQUESTED** [review]
This needs some fixes

File: types/api/xverse/appConfig.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (victorkirov) [reply types/api/xverse/appConfig.ts:10]: One last thing, looks like this type is actually not correct. The new response is a list of sections 👀

File: api/xverse.ts (hunk 3 above)
Comment (dhriaznov) [reply api/xverse.ts:313]: @m-aboelenein do we need this change?

File: types/api/xverse/appConfig.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhriaznov) [reply types/api/xverse/appConfig.ts:10]: oh, I see.. @victorkirov should we make a new PR for the API with the fixes you mentioned and merge it into `dev` and then merge those changes into `dev-k8s` or should we revert that API PR and create a new one instead and merge it into `dev-k8s`?

File: api/xverse.ts (hunk 4 above)
Comment (dhriaznov) [reply api/xverse.ts:316]: Makes sense.. we won't need the `sectionsEnabled` param then, right?

File: api/xverse.ts (hunk 3 above)
Comment (jordankzf) [reply api/xverse.ts:313]: Removed for now

File: types/api/xverse/appConfig.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  name: string;
+  url: string;
+  icon: string;
+  description: string;
+  order: number;
+  banner?: string;
+}
+
+export type DappSectionData = {
+    section: string;
```
Comment (dhriaznov) [inline types/api/xverse/appConfig.ts:11]: 4 space tabs here 😱

**Review by dhriaznov — APPROVED** [review]

**Pull request merged**